engine/lite/k8sDeploymentManager/src/main/scala/pl/touk/nussknacker/k8s/manager/K8sScalingOptionsDeterminer.scala

```diff
@@ -0,0 +1,73 @@
+package pl.touk.nussknacker.k8s.manager
+
+import net.ceedubs.ficus.Ficus
+import net.ceedubs.ficus.Ficus._
+import net.ceedubs.ficus.readers.ArbitraryTypeReader._
+import net.ceedubs.ficus.readers.ValueReader
+import pl.touk.nussknacker.k8s.manager.K8sScalingConfig._
+
+trait K8sScalingOptionsDeterminer {
+
+  def determine(parallelism: Int): K8sScalingOptions
+
+}
+
+case class K8sScalingOptions(replicasCount: Int, noOfTasksInReplica: Int)
+
+object K8sScalingOptionsDeterminer {
+
+  val defaultScalingDeterminer: K8sScalingOptionsDeterminer = new FixedReplicasCountK8sScalingOptionsDeterminer(FixedReplicasCountConfig(2))
+
+  def apply(config: Option[K8sScalingConfig]): K8sScalingOptionsDeterminer = {
+    config match {
+      case None | Some(NotDefinedConfig) => defaultScalingDeterminer
+      case Some(fixedReplicas: FixedReplicasCountConfig) => new FixedReplicasCountK8sScalingOptionsDeterminer(fixedReplicas)
+      case Some(dividingParallelism: DividingParallelismConfig) => new DividingParallelismK8sScalingOptionsDeterminer(dividingParallelism)
+    }
+  }
+
+}
+
+class FixedReplicasCountK8sScalingOptionsDeterminer(config: FixedReplicasCountConfig) extends K8sScalingOptionsDeterminer {
+
+  override def determine(parallelism: Int): K8sScalingOptions = {
+    val noOfTasksInReplica = Math.ceil(parallelism.toDouble / config.fixedReplicasCount).toInt
+    K8sScalingOptions(config.fixedReplicasCount, noOfTasksInReplica)
+  }
+
+}
+
+class DividingParallelismK8sScalingOptionsDeterminer(config: DividingParallelismConfig) extends K8sScalingOptionsDeterminer {
+
+  override def determine(parallelism: Int): K8sScalingOptions = {
+    val replicasCount = Math.ceil(parallelism.toDouble / config.tasksPerReplica).toInt
+    val noOfTasksInReplica = Math.ceil(parallelism.toDouble / replicasCount).toInt
```

Comment:
I wonder, can it be sth different than config.tasksPerReplica? And if so - do we want it?